### PR TITLE
[SIMT] Add match_all warp intrinsics

### PR DESCRIPTION
**Pull request by GaleSeLee**
Related issue = https://github.com/taichi-dev/taichi/issues/4631

1. Add match_all warp intrinsic for Cuda backend by ptx asm.
2. Add match_all unit-test.
3. Notice: there is something wrong with `nvvm:cuda_match_all_sync_i32p` intrinsic.

### CODE REVIEW

File: python/taichi/lang/simt/warp.py [hunk 1]
```diff
@@ -102,9 +102,11 @@ def match_any():
     pass
 
 
-def match_all():
-    # TODO
-    pass
+def match_all(mask, val):
+    return expr.Expr(
```
Comment (qiao-bo) [inline python/taichi/lang/simt/warp.py:106]: Can you please replace this based on the proposal here https://github.com/taichi-dev/taichi/pull/4957. Thanks!

**Review by qiao-bo — CHANGES_REQUESTED** [review]

File: python/taichi/lang/simt/warp.py (hunk 1 above)
Comment (GaleSeLee) [reply python/taichi/lang/simt/warp.py:106]: Sure

File: taichi/llvm/llvm_context.cpp [hunk 2]
```diff
@@ -387,6 +387,13 @@ std::unique_ptr<llvm::Module> TaichiLLVMContext::clone_module(
     patch_intrinsic("cuda_match_any_sync_i32",
                     Intrinsic::nvvm_match_any_sync_i32);
 
+    // LLVM 10.0.0 seems to have a bug on this intrinsic function
+    // Notice: On this intrinsic function, the third arg is a ptr
```
Comment (turbo0628) [inline taichi/llvm/llvm_context.cpp:391]: Can you comment more details about this arg?

File: tests/python/test_simt.py [hunk 3]
```diff
@@ -275,8 +275,29 @@ def test_match_any():
...
-    pass
+    a = ti.field(dtype=ti.i32, shape=32)
+    b = ti.field(dtype=ti.u32, shape=32)
+    c = ti.field(dtype=ti.u32, shape=32)
+
+    @ti.kernel
+    def foo():
+        ti.loop_config(block_dim=32)
+        for i in range(32):
+            a[i] = 0
```
Comment (qiao-bo) [inline tests/python/test_simt.py:286]: can we change this value to a non-zero? because fields are already initialized with zeros.

**Review by qiao-bo — COMMENTED** [review]
Sorry for my late reply. Could you please resolve the conflict? the rest looks good to me!

**Review by qiao-bo — APPROVED** [review]

File: tests/python/test_simt.py (hunk 3 above)
Comment (GaleSeLee) [reply tests/python/test_simt.py:286]: Sure!

**Review by qiao-bo — APPROVED** [review]

**Pull request merged**